### PR TITLE
chore(typescript): migrate core library to TS

### PR DESCRIPTION
**Pull request by EandrewJones**
Migrates entire library to TS, including support for TS in the build artifacts. Replaces Sinon with Jest for testing to resolve ESM import issues in tests.

Closes: #442

Author: Evan Jones <evan.a.jones3@gmail.com>

### CODE REVIEW

**Review by Jyyjy — CHANGES_REQUESTED** [review]

File: src/UserALEWebExtension/background.ts
Comment (EandrewJones) [inline src/UserALEWebExtension/background.ts:137]: @Jyyjy The null is likely happening here.

**Comment by EandrewJones** [comment]
@Jyyjy I had to update a few more things to get license checks to pass since the build changed. Otherwise, conflicts should be good to go. I'll let you take a look at resolve your review before merging.

**Review by Jyyjy — APPROVED** [review]

**Pull request merged**
